src/react_native/share.cljs

```diff
@@ -3,9 +3,10 @@
     ["react-native-share" :default react-native-share]))
 
 (defn open
-  ([options]
-   (open options #() #()))
-  ([options on-success on-error]
+  ([options
+    {:keys [on-success on-error]
+     :or   {on-success #()
+            on-error   #()}}]
```

Comment:
Here we're refactoring the way we receive the on-success and on-error handlers for the share-sheet. Note, the handlers are passed separately from the share-sheet options. This was done to neatly separate the data passed into the third party library `react-native-share`, though they could be mixed together if we want.

Reply:
If we're changing this, might as well not pass the callbacks and just return the promise. Now that we added promesa, we're trying to move away from this kind of callback pattern where possible.